### PR TITLE
OTC-727 Implement message signing/verifying using stored certificates

### PR DESCRIPTION
**Pull request by malinowskikam**
[https://openimis.atlassian.net/browse/OTC-727](https://openimis.atlassian.net/browse/OTC-727)

Changes:
- Updated certificate handling to store all relevant information in correct format
- Added a tool set to sign and verify messages
- Added test payload to test signature handling
- Added test for signature handling

### CODE REVIEW

File: muse_payment_adaptor/api_interface/signature.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    try:
+        public_key.verify(
+            signature,
+            message,
+            padding=__padding,
+            algorithm=__algorithm
+        )
+        return True
+    except InvalidSignature:
+        return False
```
Comment (dborowiecki) [inline muse_payment_adaptor/api_interface/signature.py:34]: Should we add a log entry explaining issue in here?

File: muse_payment_adaptor/apps.py [hunk 2]
```diff
@@ -60,16 +69,32 @@ def __load_config_config(cls, cfg):
...
+            logger.error("Error while loading a certificate", exc_info=e)
+            return None
+
+    @classmethod
+    def __load_certificate_data_as_binary(cls, path):
+        try:
+            with open(path, 'rb') as f:
+                return f.read()
+        except FileNotFoundError as e:
+            print(e)
```
Comment (dborowiecki) [inline muse_payment_adaptor/apps.py]: Please remove print()

File: muse_payment_adaptor/tests/signature.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from base64 import b64encode
+
+from muse_payment_adaptor.api_interface.signature import create_signature, verify_signature
+from muse_payment_adaptor.tests.helpers.signature_data import test_message, test_certificate, test_signature, \
+    test_wrong_signature
+from muse_payment_adaptor.apps import MusePaymentAdaptorConfig
+
+
+class SignatureTests(TestCase):
+    pass
```
Comment (dborowiecki) [inline muse_payment_adaptor/tests/signature.py]: Is this `pass` necessary?

**Review by dborowiecki — COMMENTED** [review]
Left some comments, good work.

File: muse_payment_adaptor/apps.py (hunk 2 above)
Comment (malinowskikam) [reply muse_payment_adaptor/apps.py]: fixed

File: muse_payment_adaptor/tests/signature.py (hunk 3 above)
Comment (malinowskikam) [reply muse_payment_adaptor/tests/signature.py]: fixed

File: muse_payment_adaptor/api_interface/signature.py (hunk 1 above)
Comment (malinowskikam) [reply muse_payment_adaptor/api_interface/signature.py:34]: I added a debug log entry when the signature verification fails

**Review by dborowiecki — APPROVED** [review]
Looks good

**Pull request merged**